### PR TITLE
Scan extend changes

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: samples/wifi/scan/Kconfig [hunk 1]
```diff
@@ -16,21 +16,4 @@ config WIFI_MAC_ADDRESS
...
-config WIFI_SCAN_TYPE_ACTIVE
-	bool "Active scan"
-	help
-	  Enable active scan
-
-config WIFI_SCAN_TYPE_PASSIVE
-	bool "Passive scan"
-	help
-	  Enable passive scan
-endchoice
```
Comment (krish2718) [inline samples/wifi/scan/Kconfig]: Why is this removed? The commit log itself is unclear.

File: samples/wifi/scan/src/main.c [hunk 2]
```diff
@@ -43,6 +43,12 @@ LOG_MODULE_REGISTER(scan, CONFIG_LOG_DEFAULT_LEVEL);
 
 static uint32_t scan_result;
 
+const struct wifi_scan_params tests[] = {
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:47]: Samples are to demonstrate how to use API, should only have primary usecases with inputs from Kconfig, all combinations/tests should be under `tests` usign Ztest.

File: samples/wifi/scan/prj.conf [hunk 3]
```diff
@@ -20,7 +20,7 @@ CONFIG_NET_OFFLOAD=y
 CONFIG_INIT_STACKS=y
 
 # Memories
-CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_MAIN_STACK_SIZE=4096
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf:23]: This looks like a fixup commit please fixup/squash to original commit that introduced multiple scans

**Review by krish2718 — COMMENTED** [review]
Usage of Samples Vs. Tests is my overall objection to this.

File: samples/wifi/scan/Kconfig (hunk 1 above)
Comment (D-Triveni) [reply samples/wifi/scan/Kconfig]: No idea, how it got removed. Added the changes.

File: samples/wifi/scan/src/main.c (hunk 2 above)
Comment (krish2718) [reply samples/wifi/scan/src/main.c:47]: As discussed offline lets do this in the next phase.

**Comment by D-Triveni** [comment]
Raising Draft PR to ncs. Hence closing here.
